baseplate/lib/retry.py

```diff
@@ -21,21 +23,22 @@ class RetryPolicy:
 
     """
 
-    def yield_attempts(self) -> Iterator[Optional[float]]:
+    def yield_attempts(self) -> Iterator[Tuple[Type["RetryPolicy"], Optional[float]]]:
         """Return an iterator which controls attempts.
 
-        On each iteration, the iterator will yield the number of seconds left
-        to retry, this should be used to set the timeout on the operation
-        being carried out. If there is no maximum time remaining,
-        :py:data:`None` is yielded instead.
+        On each iteration, the iterator will yield the class of the retry
+        policy and number of seconds left to retry of applicable, this
+        should be used to set the timeout on the operation being carried
+        out. If there is no maximum time remaining, :py:data:`None` is
+        yielded instead.
 
         The iterable will raise :py:exc:`StopIteration` once the operation
         should not be retried any further.
 
         """
         raise NotImplementedError
 
-    def __iter__(self) -> Iterator[Optional[float]]:
+    def __iter__(self) -> Iterator[Tuple[Type["RetryPolicy"], Optional[float]]]:
```

Comment:
This one is `Type["RetryPolicy"]` while others are `Type[RetryPolicy]` (without quotation mark).